### PR TITLE
FHB-969  acceptance tests KV

### PR DESCRIPTION
**Pull request by StuwiiDev**
- [x] Add keyvault configurations to API acceptance tests
- [x] Add Readmes to acceptance projects to explain using keyvault

### CODE REVIEW

File: src/service/referral-api/acceptance-tests/FamilyHubs.Referral.Api.AcceptanceTests/Configuration/ConfigAccessor.cs [hunk 1]
```diff
@@ -8,10 +9,14 @@ public class ConfigAccessor
...
+        var builder = new ConfigurationBuilder()
             .AddJsonFile("appsettings.json", optional: false)
-            .AddEnvironmentVariables()
-            .Build();
+            .AddUserSecrets<ConfigAccessor>()
+            .AddEnvironmentVariables();
+        
+        builder.ConfigureAzureKeyVault();
+
+        return _root ??= builder.Build();
```
Comment (stevesatdfe) [inline src/service/referral-api/acceptance-tests/FamilyHubs.Referral.Api.AcceptanceTests/Configuration/ConfigAccessor.cs:19]: _root is never defined so always null. Should we remove it or lazy initialize it?

**Review by stevesatdfe — COMMENTED** [review]
Might be worth sticking a readme in each of the acceptance test projects to help people as they might not know if they need to configure:

"AppConfiguration": {
    "KeyVaultIdentifier": "",
    "KeyVaultPrefix": ""
}

Could add something like...

## Test Setup

Use secrets to configure the variables but you can ignore:

```
"AppConfiguration": {
    "KeyVaultIdentifier": "",
    "KeyVaultPrefix": ""
}
'''

if you are running these locally as they are used by the pipeline to configure the tests.

File: src/service/referral-api/acceptance-tests/FamilyHubs.Referral.Api.AcceptanceTests/Configuration/ConfigAccessor.cs (hunk 1 above)
Comment (StuwiiDev) [reply src/service/referral-api/acceptance-tests/FamilyHubs.Referral.Api.AcceptanceTests/Configuration/ConfigAccessor.cs:19]: ye I wasn't too sure about this one, as it's static I assumed it's so it doesnt need renewing every time, I need to understand this a bit more. I'll remove it if it's not doing anything

**Review by stevesatdfe — APPROVED** [review]

File: src/shared/shared-kernel/src/fh-shared-kernel.shared-kernel/Extensions/StartupExtensions.cs [hunk 2]
```diff
@@ -7,14 +7,19 @@ namespace FamilyHubs.SharedKernel.Extensions;
 
 public static class StartupExtensions
 {
-    public static void ConfigureAzureKeyVault(this ConfigurationManager configuration)
+    public static void ConfigureAzureKeyVault(this IConfigurationBuilder configuration)
```
[the remark this answers is not in the document]
Comment (StuwiiDev) [reply src/shared/shared-kernel/src/fh-shared-kernel.shared-kernel/Extensions/StartupExtensions.cs:10]: No reason not to return it, it's a valid point. I'll add it on

**Review by Top-Cat — APPROVED** [review]

**Review by Zac-Digital — APPROVED** [review]

**Pull request merged**